language/functional_tests/src/config/global.rs

```diff
@@ -70,18 +105,49 @@ pub struct Config {
     /// A map from account names to account data
     pub accounts: BTreeMap<String, AccountData>,
     pub genesis_accounts: BTreeMap<String, Account>,
+    /// The validator set after genesis
+    pub validator_set: ValidatorSet,
 }
 
 impl Config {
     pub fn build(entries: &[Entry]) -> Result<Self> {
         let mut accounts = BTreeMap::new();
+        let mut validator_accounts = 0;
+        for entry in entries {
+            if entry.is_validator() {
+                validator_accounts += 1;
+            }
+        }
+
+        // generate a validator set with |validator_accounts| validators
+        let (validator_keys, consensus_config, network_config) =
+            ConfigHelpers::gen_validator_nodes(validator_accounts, None);
+        let validator_set = consensus_config.get_validator_set(&network_config);
+
+        // initialize the keys of validator entries with the validator set
+        // enhance type of config to contain a validator set, use it to initialize genesis
         for entry in entries {
             match entry {
                 Entry::AccountDefinition(def) => {
-                    let account_data = AccountData::new(
-                        def.balance.unwrap_or(DEFAULT_BALANCE),
-                        def.sequence_number.unwrap_or(0),
-                    );
+                    let account_data = if entry.is_validator() {
+                        validator_accounts -= 1;
+                        let validator_public_keys =
+                            validator_set.payload()[validator_accounts].clone();
```

Comment:
The counter `validator_accounts` seems funny to me. Just curious why this needs to be in reverse order?

Reply:
There's no need for it to be in reverse order; we could create a second counter that starts from 0 and goes up instead. I did it this way to reuse the first counter + make the relationship between the validator set indexes and the initial counter clear.